src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java

```diff
@@ -196,7 +196,7 @@ public static AbstractReadExecutor getReadExecutor(SinglePartitionReadCommand co
 
         // There are simply no extra replicas to speculate.
         // Handle this separately so it can record failed attempts to speculate due to lack of replicas
-        if (replicaPlan.contact().size() >= replicaPlan.liveOnly().all().size())
+        if (replicaPlan.contact().size() == replicaPlan.candidates().size())
```

Comment:
So >= is not necessary because contact should be a subset of candidates (or it should be an unavailable)? Is it more robust to use >= in case that doesn't hold due to a mistake or have an assertion for that?